src/frontends/lean/structure_cmd.cpp

```diff
@@ -490,9 +490,24 @@ struct structure_cmd_fn {
         } else {
             expr new_tmp = elab(mk_arrow(in_header ? parent : mk_as_is(parent), tmp));
             parent       = copy_tag(parent, expr(binding_domain(new_tmp)));
+            unsigned i   = static_cast<unsigned>(&parent - &m_parents[0]);
+
```

Comment:
This cast and pointer arithmetic looks fishy to me. Could you please clarify what is going on here? It is not obvious that `parent` is in `m_parents`.

Reply:
Yeah, this is unnecessarily clever. I will change the code so that the function is passed `i` instead of `parent`.